packages/uhk-web/src/app/store/reducers/user-configuration.ts

```diff
@@ -184,14 +214,28 @@ export function reducer(state = initialState, action: Action & { payload?: any }
                 return macro.id !== action.payload.id && macro.name === name;
             });
 
-            changedUserConfiguration.macros = state.macros.map((macro: Macro) => {
-                macro = Object.assign(new Macro(), macro);
-                if (!duplicate && macro.id === action.payload.id) {
-                    macro.name = name;
-                }
+            if (duplicate) {
+                break;
+            }
 
-                return macro;
-            });
+            const index = state.macros.findIndex(macro => macro.id === action.payload.id);
+            const macroToRename = state.macros[index];
+            const newMacro = Object.assign(new Macro(), macroToRename, { name });
+            const newMacros: Macro[] = [];
```

Comment:
why do not use the `addMacro` function here? If I see well it is the same code.

Reply:
Yes, just need to add a function which checks for the old item, which shouldn't be added.